pkg/deploy/config.go

```diff
@@ -0,0 +1,101 @@
+package deploy
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+
+	"gopkg.in/yaml.v2"
+)
+
+// Config represents configuration object for deployer tooling
+type Config struct {
+	RPS           []RPConfig    `yaml:"rps"`
+	Configuration Configuration `yaml:"configuration"`
+}
+
+// RPConfig represents individual RP configuration
+type RPConfig struct {
+	// Name is unique identifier of RP
+	Name              string        `yaml:"name"`
+	Location          string        `yaml:"location"`
+	ResourceGroupName string        `yaml:"resourceGroupName"`
+	SubscriptionID    string        `yaml:"subscriptionId"`
+	Configuration     Configuration `yaml:"configuration"`
+}
+
+// Configuration represents configuration structure
+type Configuration struct {
+	AdminAPIClientCertCommonName string `yaml:"adminApiClientCertCommonName"`
+	AdminAPICaBundle             string `yaml:"adminApiCaBundle"`
+	KeyvaultPrefix               string `yaml:"keyvaultPrefix"`
+	RPServicePrincipalID         string `yaml:"rpServicePrincipalId"`
+	FPServicePrincipalID         string `yaml:"fpServicePrincipalId"`
+	ExtraKeyvaultAccessPolicies  string `yaml:"extraKeyvaultAccessPolicies"`
+	RpImage                      string `yaml:"rpImage"`
+	RpMode                       string `yaml:"rpMode"`
+	SSHPublicKey                 string `yaml:"sshPublicKey"`
+	VMSSName                     string `yaml:"vmssName"`
+	MDMFrontendURL               string `yaml:"mdmFrontendUrl"`
+	DatabaseAccountName          string `yaml:"databaseAccountName"`
+	ExtraCosmosDBIPs             string `yaml:"extraCosmosDBIPs"`
+	PullSecret                   string `yaml:"pullSecret"`
+	DomainName                   string `yaml:"domainName"`
+	MDSDConfigVersion            string `yaml:"mdsdConfigVersion"`
+	RPImageAuth                  string `yaml:"rpImageAuth"`
+	MDSDEnvironment              string `yaml:"mdsdEnvironment"`
+}
+
+// GetConfig return RP configuration from config.yaml file in the execution root
+// path. Location of the file can be provided using DEPLOy_CONFIG variable.
+func GetConfig(region string) (*RPConfig, error) {
+	configFile := "config.yaml"
+	if os.Getenv("DEPLOY_CONFIG") != "" {
+		configFile = os.Getenv("DEPLOY_CONFIG")
+	}
```

Comment:
why? can we avoid this?

Reply:
I suspect as soon as we hit CI or multiple config files - we will need this.